pandas/tools/pivot.py

```diff
@@ -434,12 +437,44 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
     data.update(zip(rownames, index))
     data.update(zip(colnames, columns))
 
+    # Sanity checks
+    if values is None and callable(aggfunc):
+        raise ValueError("Custom `aggfunc` functions require `values`"
+                         "to be specified")
+
+    if isinstance(aggfunc, str):
+        if aggfunc not in ['pct', 'pct_rows', 'pct_cols']:
+            raise ValueError('Not a valid `aggfunc` argument')
+
     if values is None:
         df = DataFrame(data)
         df['__dummy__'] = 0
         table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
                                aggfunc=len, margins=margins, dropna=dropna)
-        return table.fillna(0).astype(np.int64)
+
+        if isinstance(aggfunc, str):
```

Comment:
you mean instead of `if aggfunc == 'pct':`, define all those transformations into functions and then call `table = normalization_func(aggfunc)(table)` ?